cmd/readiness/main.go

```diff
@@ -45,59 +44,38 @@ func init() {
 // Additionally if the previous check hasn't returned 'true' the "deadlock" case is checked to make sure the Agent is
 // not waiting for the other members.
 func isPodReady(conf config.Config) bool {
-	fd, err := os.Open(conf.HealthStatusFilePath)
+	healthStatus, err := conf.HealthStatusProvider.HealthStatus()
 	if err != nil {
-		logger.Warn("No health status file exists, assuming the Automation agent is old")
-		return true
-	}
-	defer fd.Close()
-
-	health, err := readAgentHealthStatus(fd)
-	if err != nil {
-		logger.Errorf("Failed to read agent health status file: %s", err)
-		// panicking allows to see the problem in the events for the pod (kubectl describe pod ..)
-		panic("Failed to read agent health status file: %s")
+		panic(err)
```

Comment:
why panic here? Maybe we can log an error and return false? It looks like we have a mix of styles of control flow.

Reply:
`panic` is an indication of the error - this is shown when you call `kubectl describe pod`, while logging writes the data into the `readiness.log` and no one usually cares about that file.
Happy to discuss this though 👍

Reply:
just an example: this is how the `kubectl describe` looks if the readiness probe fails: 

```
...
  Normal   Started    98s               kubelet            Started container mongodb-agent-readinessprobe
  Normal   Pulling    98s               kubelet            Pulling image "registry.hub.docker.com/library/mongo:4.2.6"
  Normal   Pulled     82s               kubelet            Successfully pulled image "registry.hub.docker.com/library/mongo:4.2.6" in 16.479752s
  Normal   Created    81s               kubelet            Created container mongod
  Normal   Started    81s               kubelet            Started container mongod
  Normal   Pulling    81s               kubelet            Pulling image "quay.io/mongodb/mongodb-agent:10.29.0.6830-1"
  Normal   Pulled     69s               kubelet            Successfully pulled image "quay.io/mongodb/mongodb-agent:10.29.0.6830-1" in 12.4341038s
  Normal   Created    67s               kubelet            Created container mongodb-agent
  Normal   Started    67s               kubelet            Started container mongodb-agent
  Warning  Unhealthy  3s (x6 over 53s)  kubelet            Readiness probe failed: panic: read /var/log/mongodb-mms-automation/healthstatus/agent-health-status.json: file already closed

goroutine 1 [running]:
main.main()
  /build/main.go:218 +0x2d3

```